drivers/net/ethernet/renesas/rswitch.c

```diff
@@ -3804,7 +3822,82 @@ static void rswitch_fwd_init(struct rswitch_private *priv)
 
 	/* Enable SC-Tag filtering mode for VLANs */
 	rs_write32(BIT(1), priv->addr + FWGC);
+
+#if IS_ENABLED(CONFIG_RENESAS_FWD_MON)
+	// CPU mirroring
+	rs_write32(priv->mon_chain | (RSWITCH_HW_NUM_TO_GWCA_IDX(priv->gwca.index) << 16),
+	   priv->addr + FWCMPTC);
+#endif
+}
+
+#if IS_ENABLED(CONFIG_RENESAS_FWD_MON)
+static int rswitch_mon_ndev_allocate(struct rswitch_private *priv)
```

Comment:
I can see that there are lots of functions that allocate ndev in the rswitch2 code. May I ask you to consider to factor out similar bits into some helper function? Because it getting out of hand.

Or at least add TODO comment there.

Reply:
The code was completely reworked and I tried to reuse the code for the maximum.